packages/excalidraw/components/App.tsx

```diff
@@ -2146,11 +2146,8 @@ class App extends React.Component<AppProps, AppState> {
   };
 
   public dismissLinearEditor = () => {
-    setTimeout(() => {
-      this.setState({
-        editingLinearElement: null,
-      });
-    });
+    this.state.editingLinearElement &&
+      this.actionManager.executeAction(actionFinalize);
```

Comment:
I don't think this is an identical effect change to what it was originally. Making this change introduced the regression @dwelle was experiencing with loss of selection. More specifically, idiomatically dismissing the linear editor is not the exact same as finalizing as we use the action currently.

Reply:
This is a great point, I'm now wondering if using `actionFinalize` to dismiss the linear editor is a bad idea alltogether.
If I change the `dismissLinearEditorFunction` to:
```
  public dismissLinearEditor = () => {
    this.setState({
      editingLinearElement: null,
    });
  };
```
things work perfectly (I don't even need to modify`actionFinalize` this way) AND it also solves issue 2 (this way, after the linear editor is dismissed the user can edit the line using it's transform handels)

Reply:
If you reverse these and implement your specific changes it will work just fine and I bet the tests are also going to pass. Please note that the snapshots for the tests will need to be regenerated (and the new snapshots be committed) in order to pass CI. Just run `yarn test:update` after you're done with your changes.